### ISSUE TITLE
Feature or bug? When Authorization header set to false login fails

### ISSUE THREAD

**Issue by greenermoose** [strapi/strapi#8953]
## Bug report

When Authorization header set to false unable to login

### Describe the bug

I'm trying to get nuxt/auth working with strapi 3.4.0. On a login before a user is authenticated, it appears that nuxt/auth is setting the Authorization header to false. Strapi complains with

```
{
"statusCode": 401,
"error": "Unauthorized",
"message": "Invalid authorization header format. Format is Authorization: Bearer [token]"
}
```

### Steps to reproduce the behavior

HTTP POST to /auth/local with JSON body a valid login and Authorization header set to false. Not sure why nuxt/auth is setting this header, but it is reasonable since we are not logged in yet and trying to login.

Strapi complains with

```
{
"statusCode": 401,
"error": "Unauthorized",
"message": "Invalid authorization header format. Format is Authorization: Bearer [token]"
}
```

### Expected behavior

I was expecting to be able to login even with an Authorization header set to false. Since I'm hitting the login REST endpoint, I'd expect any Authorization headers to be ignored, or have the option of ignoring the Authorization header. Couldn't find how to do that in the strapi docs.

### Code snippets

Here is nuxt code that triggers the problem :

```
        let response = await this.$auth.loginWith("local", {
          data: {
            identifier: loginInfo.email,
            password: loginInfo.password
          }
        });
```

Here is code that works:

```
        let response = await this.$auth.loginWith("local", {
          data: {
            identifier: loginInfo.email,
            password: loginInfo.password
          },
          headers: {
            "Authorization": ""
          }
        });
```

### System

- Node.js version: 14.15.1
- Strapi version: 3.4.0
- Nuxt: 2.14.12
- Nuxt/auth: 5.0

**Comment by derrickmehaffy** [comment]
This one is difficult, there is a simple "workaround" solution which is what we intended to be done... simply don't send the header, which makes the request "public".

By nature, if the authorization header is present, the request is not public and thus the token is validated.

The main issue is **when** we check for this header. Under your assumption, the header is checked in the same function as the auth request and not before. I think that's the case but I'll have to look at the code to confirm. And if so, we could add a cause to look for the `authorization: false` boolean.

Generally its not a good practice to assign the same value as multiple types.

I'll ping the backend @strapi/maintainers to get their thoughts and mark as an unconfirmed bug until they respond (though with the holidays this might not be til Jan).

---

Tldr: don't send the header

**Comment by greenermoose** [comment]
Thanks for the quick response, @derrickmehaffy! I think the nuxt/auth maintainers are working on a fix on their side, taking your advice and not setting the Authorization header. In the meantime, hope this issue report helps anyone else who might run into this problem. Took me a while to track down why the login was failing during what should have a been a quick test of using nuxt as a front end for strapi!

**Comment by derrickmehaffy** [comment]
Functionally not a bug on our side so going to close
